### ISSUE TITLE
MODX 3 Language during installation does not match manager language

### ISSUE THREAD

**Issue by Ruslan-Aleev** [modxcms/revolution#14694]
### Summary
During the installation, I chose English, but in the form of login and manager - the language is already Russian.

### Expected behavior
The language should match the one selected during the installation process.

### Environment
MODX 3

**Comment by Mark-H** [comment]
I consider that to be an amazing feature, not a bug ;) 

The login/manager automatically selects the most appropriate language based on your browser language. To change it you can use the language selector in the bottom of the login screen and in the user menu in the manager.

**Comment by Ruslan-Aleev** [comment]
And if I need a language that is not my browser? Maybe I'm installing not for myself?

**Comment by Ruslan-Aleev** [comment]
And why does the system consider itself smarter =) I want the language that I chose.

**Comment by Mark-H** [comment]
Like I said, you can still choose with the language selector. It's just per user instead of a global setting.

**Comment by Ruslan-Aleev** [comment]
I understand, but you force users to take meaningless steps, this is bad for UX.
We can add another 2-3 switches that have little effect, but they can be clicked =)

**Comment by Mark-H** [comment]
Someone with a Russian browser is likely to speak Russian, and the best UX would be the manager to be in Russian. If they don't want Russian (why do they use a browser set to Russian?), the language selection is just one click away and their preference will be remembered for future logins.

Someone with a Dutch browser is likely to speak Dutch, and the best UX would be the manager to be in Dutch. If they don't want Dutch (why do they use a browser set to Dutch?), the language selection is just one click away and their preference will be remembered for future logins.

Wether MODX was installed in Russian, Dutch, English or Swedish doesn't matter - the _user_ experience is about what best fits a _specific user_ which is not dependant on an installation language. 

That you happen to use a browser set to Russian but want English does not mean that is true for the average user. 

In none of those cases the user is **forced** to to take meaningless steps. If anything, they are **saved the trouble of having to create a user setting** to override the manager language if they want something different. The system takes a best guess, based on the browser language, which is a pretty good indication. 

I only see much improved UX improvements there. A very important feature, choosing the manager language, is now personalised to the user. 

It's a change you will need to get used to, but give it some time.

**Comment by Ruslan-Aleev** [comment]
Yes, that’s understandable. But it would be correct in the future that the language of the installer should be taken based on the language of the browser. Now, in my opinion, this is inconsistent.

**Comment by Ruslan-Aleev** [comment]
Also, the inconsistency of the language of the login form and the one selected in the manager panel:
If you change the language to another and exit the admin panel, then the login form will display your system language (and not the one selected before exiting the panel), i.e. now you need to select the language again.

**Comment by Ibochkarev** [comment]
The login still behaves as reported, and the last note in the thread is already how setup starts.

`SecurityLoginManagerController::handleLanguageChange()` picks the manager language from the request, then the session, then `Accept-Language`, then `en`. It does not read the language chosen in setup. `setup/includes/new.install.php` writes `cultureKey` only when that language is not `en`. It does not write `manager_language`.

`modInstallLexicon::getLanguage()` already follows the browser when the `modx_setup_language` cookie is absent. It walks `Accept-Language` and keeps the first tag that matches a directory under `setup/lang/`. `setup/controllers/language.php` passes that value as `$current`, and `language.tpl` checks that radio and loads the lexicon in that language.

A Russian browser therefore opens setup in Russian and the login in Russian. Choosing English on the language step changes the installer and, when the choice is not English, the site `cultureKey`. The login stays with the browser until someone picks a language there, and that choice is stored in the session.

That is the split from the thread: the manager language belongs to the browser, and setup now starts from the same header. Nothing further to change on this issue.
